### PR TITLE
Feat: Table footer in `eds-core-react` & `eds-data-grid-react`

### PR DESCRIPTION
**Pull request by zulu-eq-bouvet**
## What does this pull request change?

**`feat`:**  `add Table footer component to eds-core-react`
- Implemented the Table footer component with optional sticky property for a fixed footer.
- Updated `Table.docs.mdx` to include documentation for the Table footer.
- Added Table footer to the storybook to showcase various use cases.
- Updated test suites to cover edge cases and ensure proper functionality of the Table footer.

**`feat:`** `add Table Footer component to eds-data-grid-react`
- Added the possibility to show a footer in the data grid.
- Enabled the option to make the footer fixed.
- Added `footerClass` and `footerStyle` as component props for custom styling.
- Updated docs and stories to include the new footer functionality.
- Updated test suites to cover edge cases and ensure proper functionality of the Table Footer.

**`chore`:** extract `Resizer` and `TableCell` components

- `Resizer` is now a separate module to enhance reusability and scalability.
- Renamed `Cell` to `TableCell` and moved it to a separate module for better reusability and scalability.

## Why is this pull request needed?
This pull request adds support for displaying the table footer in both **`eds-core-react`** and **`eds-data-grid-react`**. Additionally, it introduces a property that allows the footer to be fixed aka **`sticky`**.

**`Note:`** We do not have a design for the Table footer component, so it currently inherits header cell token values. This can be modified in a future update.

### CODE REVIEW

**Review by oddvernes — APPROVED** [review]
Looks all good to me, will wait for feedback from @yusijs as well 👍

**Review by yusijs — APPROVED** [review]
I forgot to approve when I looked earlier :D But lgtm as well!

**Pull request merged**
